CMake cache creation breaks when using HPX with mimalloc

## Expected Behavior

To build the cmake cache correctly

## Actual Behavior

When building the cmake cache for my application inside this command:
```
find_package(HPX CONFIG REQUIRED)
```
A compile of a small test program is launched, which injects a symbol with 
```
/INCLUDE:mi_version
```
during linking, which then generates a missing symbol error and is breaking the build.

## Steps to Reproduce the Problem

1. Build HPX with mimalloc enabled
2. Use that HPX version (Windows build)

## Specifications

- HPX Version: latest stable
- Platform (compiler, OS): MSVC, latest, Windows 10

Error output in cmake configuration phase when entering `find_package(HPX CONFIG REQUIRED)` :
```
1> [CMake] -- Using mimalloc allocator.
1> [CMake] -- Found Boost: C:/Arcanima/AC_DOWNLOAD/vcpkg/installed/x64-windows/include (found suitable version "1.73.0", minimum required is "1.61")  
1> [CMake] -- Found Boost: C:/Arcanima/AC_DOWNLOAD/vcpkg/installed/x64-windows/include (found suitable version "1.73.0", minimum required is "1.61") found components: filesystem 
1> [CMake] -- The C compiler identification is MSVC 19.26.28806.0
1> [CMake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe
1> [CMake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe - broken
1> [CMake] CMake Error at C:/Program Files (x86)/CMake/share/cmake-3.17/Modules/CMakeTestCCompiler.cmake:60 (message):
1> [CMake]   The C compiler
1> [CMake] 
1> [CMake]     "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe"
1> [CMake] 
1> [CMake]   is not able to compile a simple test program.
1> [CMake] 
1> [CMake]   It fails with the following output:
1> [CMake] 
1> [CMake]     Change Dir: C:/Arcanima/AC_BUILD/sim/0.0.1.0/RelWithDebInfo/CMakeFiles/CMakeTmp
1> [CMake]     
1> [CMake]     Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_e2c5e && [1/2] Building C object CMakeFiles\cmTC_e2c5e.dir\testCCompiler.c.obj
1> [CMake]     [2/2] Linking C executable cmTC_e2c5e.exe
1> [CMake]     FAILED: cmTC_e2c5e.exe 
1> [CMake]     cmd.exe /C "cd . && "C:\Program Files (x86)\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\cmTC_e2c5e.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cmTC_e2c5e.dir\testCCompiler.c.obj  /out:cmTC_e2c5e.exe /implib:cmTC_e2c5e.lib /pdb:cmTC_e2c5e.pdb /version:0.0  /machine:x64 /INCLUDE:mi_version  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D C:\Arcanima\AC_BUILD\sim\0.0.1.0\RelWithDebInfo\CMakeFiles\CMakeTmp && powershell -noprofile -executionpolicy Bypass -file C:/Arcanima/AC_DOWNLOAD/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Arcanima/AC_BUILD/sim/0.0.1.0/RelWithDebInfo/CMakeFiles/CMakeTmp/cmTC_e2c5e.exe -installedDir C:/Arcanima/AC_DOWNLOAD/vcpkg/installed/x64-windows/debug/bin -OutVariable out""
1> [CMake]     LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cmTC_e2c5e.dir\testCCompiler.c.obj /out:cmTC_e2c5e.exe /implib:cmTC_e2c5e.lib /pdb:cmTC_e2c5e.pdb /version:0.0 /machine:x64 /INCLUDE:mi_version /debug /INCREMENTAL /subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\cmTC_e2c5e.dir/intermediate.manifest CMakeFiles\cmTC_e2c5e.dir/manifest.res" failed (exit code 1120) with the following output:
1> [CMake]     LINK : error LNK2001: unresolved external symbol mi_version
1> [CMake]     cmTC_e2c5e.exe : fatal error LNK1120: 1 unresolved externals
1> [CMake]     ninja: build stopped: subcommand failed.
1> [CMake]     
1> [CMake]     
1> [CMake] 
1> [CMake]   
1> [CMake] 
1> [CMake]   CMake will not be able to correctly generate this project.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   C:/Arcanima/AC_INSTALL/hpx/stable/RelWithDebInfo/lib/cmake/HPX/HPXConfig.cmake:106 (enable_language)
1> [CMake]   C:/Arcanima/AC_DOWNLOAD/vcpkg/scripts/buildsystems/vcpkg.cmake:329 (_find_package)
1> [CMake]   repos/sim/CMakeLists.txt:119 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "C:/Arcanima/AC_BUILD/_MSVC/BUILD/Windows-X64-RelWithDebInfo/CMakeFiles/CMakeOutput.log".
1> 'cmd.exe' '/c ""C:\Program Files (x86)\CMake\bin\cmake.exe"  -G "Ninja" -DCMAKE_INSTALL_PREFIX:PATH="C:\Arcanima\RelWithDebInfo" -DVCPKG_VERBOSE:BOOL="True"  -DCMAKE_TOOLCHAIN_FILE="C:/Arcanima/AC_DOWNLOAD/vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Arcanima\AC_SB" 2>&1"' execution failed with error: ''cmd.exe' '/c ""C:\Program Files (x86)\CMake\bin\cmake.exe"  -G "Ninja" -DCMAKE_INSTALL_PREFIX:PATH="C:\Arcanima\RelWithDebInfo" -DVCPKG_VERBOSE:BOOL="True"  -DCMAKE_TOOLCHAIN_FILE="C:/Arcanima/AC_DOWNLOAD/vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Arcanima\AC_SB" 2>&1"' returned with exit code: 1'.
```

--- a/cmake/HPX_SetupAllocator.cmake
+++ b/cmake/HPX_SetupAllocator.cmake
@@ -52,7 +52,9 @@ if(NOT TARGET hpx_dependencies_allocator)
       )
 
       if(MSVC)
-        hpx_add_link_flag_if_available(/INCLUDE:__tcmalloc)
+        target_compile_options(
+          hpx_dependencies_allocator INTERFACE /INCLUDE:__tcmalloc
+        )
       endif()
       set(_use_custom_allocator TRUE)
     endif()
@@ -83,7 +85,9 @@ if(NOT TARGET hpx_dependencies_allocator)
       target_link_libraries(hpx_dependencies_allocator INTERFACE mimalloc)
       set(hpx_MALLOC_LIBRARY mimalloc)
       if(MSVC)
-        hpx_add_link_flag_if_available(/INCLUDE:mi_version)
+        target_compile_options(
+          hpx_dependencies_allocator INTERFACE /INCLUDE:mi_version
+        )
       endif()
       set(_use_custom_allocator TRUE)
     endif()
@@ -96,7 +100,9 @@ if(NOT TARGET hpx_dependencies_allocator)
         hpx_error(${allocator_error})
       endif()
       if(MSVC)
-        hpx_add_link_flag_if_available(/INCLUDE:__TBB_malloc_proxy)
+        target_compile_options(
+          hpx_dependencies_allocator INTERFACE /INCLUDE:__TBB_malloc_proxy
+        )
       endif()
       target_link_libraries(
         hpx_dependencies_allocator INTERFACE ${TBBMALLOC_LIBRARY}

--- a/cmake/templates/HPXConfig.cmake.in
+++ b/cmake/templates/HPXConfig.cmake.in
@@ -103,8 +103,6 @@ if(NOT HPX_CMAKE_LOGLEVEL)
   set(HPX_CMAKE_LOGLEVEL "WARN")
 endif()
 
-enable_language(C)
-
 hpx_check_cmake_build_type()
 hpx_check_compiler_compatibility()
 hpx_check_boost_compatibility()

